netbout-web/src/main/java/com/netbout/rest/bout/TkAttach.java

```diff
@@ -137,12 +138,31 @@ public Response act(final Request req) throws IOException {
     }
 
     /**
-     * Extracts file name.
+     * Reads the filename either from the name field or from the uploaded file.
+     * @param form From
+     * @param file File
+     * @return String name
+     * @throws IOException If fails
+     */
+    private String filename(final Smart form, final Request file)
+        throws IOException {
+        String name = null;
```

Comment:
@jhyle you could remove the `null` and make `name` final if you would go with such solution:
```
final String name;
Iterable<Request> reqs = form.part("name");
if (reqs.hasNext()) {
    name = IOUtils.toString.....;
} else {
   name = this.nameFromFile....;
}
```